src/components/SettingsButton/test.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { screen, fireEvent, waitFor } from '@testing-library/react';
+
+import { renderWithClient } from 'testUtils';
+import { SettingsButton } from '.';
+
+describe('SettingsButton', () => {
+    test('renders', async () => {
+        const { rendered } = renderWithClient(<SettingsButton />);
+        const button = rendered.container.querySelector('.settings_button');
+        if (button) fireEvent.click(button);
+        await waitFor(() => screen.getByText('Widget Settings'));
+    });
+    test('has a toggle which will change on click', async () => {
+        const { rendered } = renderWithClient(<SettingsButton />);
+        const button = rendered.container.querySelector('.settings_button');
+        if (button) fireEvent.click(button);
+        await waitFor(() => {
+            screen.getByText('Widget Settings');
+            const toggle = rendered.container.querySelectorAll('.toggle')[0] as HTMLInputElement;
+            if (toggle) {
```

Comment:
This will pass if there's no toggle, I think just `expect(toggle.checked).toBe(false)` without the `if` should be fine since that will fail if the toggle isn't defined.